### PR TITLE
Intra process tracepoints

### PR DESCRIPTION
**Pull request by cwecht**
This PR adds support for the tracepoint for intra process communication introduced in

- https://github.com/ros2/rclcpp/pull/2091
- https://github.com/ros2/ros2_tracing/pull/30

Please note, that this PR is based on https://github.com/safe-ros/ros2_profiling/pull/23. 

This implementation seems to be a bit rough to me, but it should suffice as a proof of concept.

### CODE REVIEW

**Comment by mjcarroll** [comment]
I like this, but will probably defer on merging until the upstream intraprocess trace points land (I think they are relatively close).

**Comment by cwecht** [comment]
@mjcarroll  As the intra process trace points are now merged, this PR can now considered for merging, right?

File: ros2profile/ros2profile/data/constants.py [hunk 1]
```diff
@@ -43,3 +43,12 @@
...
 RMW_TAKE = 'ros2:rmw_take'
 DDS_READ = 'dds:read'
+
+IPB_TO_SUBSCRIPTION = 'ros2:IPB_to_subscription'
+BUFFER_TO_TYPED_IPB = 'ros2:buffer_to_typedIPB'
+CONSTRUCT_RINGBUFFER = 'ros2:construct_ring_buffer'
+RINGBUFFER_ENQUEUE = 'ros2:ring_buffer_enqueue'
+RINGBUFFER_DEQUEUE = 'ros2:ring_buffer_dequeue'
+RINGBUFFER_CLEAR = 'ros2:ring_buffer_clear'
+RCLCPP_INTRA_PUBLISH = 'ros2:rclcpp_intra_publish'
```
Comment (christophebedard) [inline ros2profile/ros2profile/data/constants.py]: some of the tracepoints were renamed, and all of them now use the `rclcpp_` prefix: https://github.com/ymski/ros2_tracing/blob/944fbfbee9376414d55069b760b2122c1e2734c8/tracetools_trace/tracetools_trace/tools/tracepoints.py#L46-L51

File: ros2profile/ros2profile/data/convert/ctf.py [hunk 2]
```diff
@@ -69,6 +69,11 @@
     "symbol": str,
     "state_machine": int,
     "gid": lambda x: list(map(int, x)),
+    "IPB": int,
+    "capacity": int,
+    "index": int,
+    "accumulated_data": int,
+    "overwriting_occurred": bool,
```
Comment (christophebedard) [inline ros2profile/ros2profile/data/convert/ctf.py]: some field names were renamed: https://github.com/ymski/ros2_tracing/blob/944fbfbee9376414d55069b760b2122c1e2734c8/tracetools/include/tracetools/tracetools.h#L537-L616

Comment (cwecht) [reply ros2profile/ros2profile/data/convert/ctf.py]: Fixed.

File: ros2profile/ros2profile/data/constants.py (hunk 1 above)
Comment (cwecht) [reply ros2profile/ros2profile/data/constants.py]: Fixed.

**Review by mjcarroll — APPROVED** [review]

**Pull request merged**